public/app/plugins/datasource/influxdb/datasource.mock.ts

```diff
@@ -0,0 +1,54 @@
+type FieldsDefinition = {
+  name: string;
+  // String type, usually something like 'string' or 'float'.
+  type: string;
+};
+type Measurements = { [measurement: string]: FieldsDefinition[] };
+type FieldReturnValue = { text: string };
+
+/**
+ * Datasource mock for influx. At the moment this only works for queries that should return measurements or their
+ * fields and no other functionality is implemented.
+ */
+export class InfluxDatasourceMock {
+  constructor(private measurements: Measurements) {}
+  metricFindQuery(query: string) {
+    if (isMeasurementsQuery(query)) {
+      return this.getMeasurements();
+    } else {
+      return this.getMeasurementFiels(query);
+    }
+  }
+
+  private getMeasurements(): FieldReturnValue[] {
+    return Object.keys(this.measurements).map(key => ({ text: key }));
+  }
+
+  private getMeasurementFiels(query: string): FieldReturnValue[] {
+    const match = query.match(/SHOW FIELD KEYS FROM \"(.+)\"/);
```

Comment:
Should this be `getMeasurementFields`?

Reply:
Not sure what you mean. Do you have some other name in mind for the function or do you mean the regex does not seem to match the name of the method?